src/main/java/com/alvarium/annotators/vulnerability/GoModHandler.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright 2023 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.alvarium.annotators.vulnerability;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class GoModHandler implements PackageFileHandler {
+    final private File file;
+
+    protected GoModHandler(String dir) {
+        this.file = new File(dir + "/" + "go.mod");
+    }
+
+    @Override
+    public String getFileName() {
+        return this.file.getName();
+    }
+
+    @Override
+    public Map<String, String> getPackages() throws VulnerabilityException {
+        Map<String, String> packages = new HashMap<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String line = null;
+            boolean inRequireBlock = false;
+            while ((line = reader.readLine()) != null) {
+                line = line.trim();
```

Comment:
Instead of having to declare this `line` var outside of its scope which is inside the while loop and also explicitly setting it as `null`. Perhaps using a do-while loop would be cleaner ? 

```java
do {
    String line = reader.readLine();
    if (line == null) break; // probably not needed, but defends against empty files 
    // ... rest of logic here
} while(line != null)
```

Reply:
Won't I still need to declare the line outside of the do while to be resolve in the while condition?
```java
String line;
do {
    line = reader.readLine();
    if (line == null) break; // probably not needed, but defends against empty files 
    // ... rest of logic here
} while(line != null)
```